explorer/explorer_android.go

```diff
@@ -80,16 +72,73 @@ func initLib() {
 	}
 }
 
-type File struct {
-	obj                    jni.Object
-	sharedBuffer           jni.Object
-	sharedBufferLen        int
-	isWritable, isReadable bool
-	isClosed               bool
+func (e *Explorer) listenEvents(evt event.Event) {
+	if evt, ok := evt.(app.ViewEvent); ok {
+		e.view = evt.View
+		e.initLib()
+	}
+}
+
+func (e *Explorer) exportFile(name string) (io.WriteCloser, error) {
+	go func() {
+		jni.Do(jni.JVMFor(app.JavaVM()), func(env jni.Env) error {
+			return jni.CallVoidMethod(env, e.libObject, e.explorer.exportFile,
+				jni.Value(e.view),
+				jni.Value(jni.JavaString(env, strings.TrimPrefix(strings.ToLower(filepath.Ext(name)), "."))),
+				jni.Value(e.id),
+			)
+		})
+	}()
+
+	file := <-e.result
+	if file.error != nil {
+		return nil, file.error
+	}
+	return file.file.(io.WriteCloser), nil
 }
 
-func (f *File) Read(b []byte) (n int, err error) {
-	if f == nil || !f.isReadable || f.isClosed {
+func (e *Explorer) importFile(extensions ...string) (io.ReadCloser, error) {
+	for i, ext := range extensions {
+		extensions[i] = mime.TypeByExtension(ext)
+	}
+
+	mimes := strings.Join(extensions, ",")
+	if len(mimes) == 0 {
+		mimes = ""
+	}
```

Comment:
I don't think this check is necessary. [As far as I can tell](https://go.dev/play/p/MfAqEw_UCi0), it's a no-op. Am I missing something?

Reply:
Yes, it must be removed. I forget to remove it, while debugging. 😂

Reply:
That was fixed on `explorer: [android] remove useless string`.